### DISCUSSION TITLE
Auto-close on mobile nav when route selected

### DISCUSSION THREAD

**Discussion by pkennedytx1** [heroui-inc/heroui#1094] [Feedback]
The mobile nav should have a prop that allows the nav to be close when a route is selected, it results in extra clicks for a user and disrupts the flow. I would imagine this as being a prop passed to Navbar.Collapse since the direct children would be the ones that the component would be listening to. I have worked around this by creating a simulated click to press the toggle after select.

**Comment by Kismet333** [comment]
I +1 this. When using the Navbar.Collapse view the dropdown should be closed upon the user making a selection, or as @pkennedytx1 mentioned, have a prop so that its behavior can be controlled. Using this Navbar with a mobile phone makes it hard for the user to even tell that their click worked because the dropdown is covering the page. I could see a user getting frustrated and thinking the website does not work. Best case scenario the user has to do unnecessary clicks to close the menu themselves.

**Comment by HackerShohag** [comment]
@pkennedytx1 @Kismet333, is there any update about this fix or how can this be solved?

**Reply by pkennedytx1** [reply]
@HackerShohag I tried to use a clickEvent so solve this issue, it works on the browser in a smaller screen width, but, does not work on a phone, also attempted to utilize onPress to no avail.

**Reply by HackerShohag** [reply]
@pkennedytx1 
See the solutions bellow, it provides a complete structure but requires client side rendering.
https://github.com/nextui-org/nextui/discussions/1267

**Comment by pkennedytx1** [comment]
I figured out a dirty solution as well:

`onTouchEnd={() => {
      setIsMenuOpen(false);
      push('/services')
  }}`
  
  You can use the Javascript listener to capture the touch event and the push the user to the correct route.

**Reply by pkennedytx1** [reply]
This is an attribute being used inside of the Link Component from nextui
